### DISCUSSION TITLE
Consider allowing retrieval of file metadata in Sftp utilities

### DISCUSSION THREAD

**Discussion by sleberknight** [kiwiproject/kiwi#583] [Ideas]
Currently we can get the file _names_ but we cannot get the additional file metadata such as file size, creation and modification times, etc. JSch provides this information via the `LsEntry` (inner class in `ChannelSftp`). In addition to file name it contains "longname" (not sure what that is...) as well as an instance of `SftpATTRS` which contains metadata as described in the comment at the top of the class, which I'm reproducing here:

```java
/*
  uint32   flags
  uint64   size           present only if flag SSH_FILEXFER_ATTR_SIZE
  uint32   uid            present only if flag SSH_FILEXFER_ATTR_UIDGID
  uint32   gid            present only if flag SSH_FILEXFER_ATTR_UIDGID
  uint32   permissions    present only if flag SSH_FILEXFER_ATTR_PERMISSIONS
  uint32   atime          present only if flag SSH_FILEXFER_ACMODTIME
  uint32   mtime          present only if flag SSH_FILEXFER_ACMODTIME
  uint32   extended_count present only if flag SSH_FILEXFER_ATTR_EXTENDED
  string   extended_type
  string   extended_data
    ...      more extended data (extended_type - extended_data pairs),
             so that number of pairs equals extended_count
*/
```

So, we could get a bunch of additional information that might be useful (for example the size and the modified time, which would be useful in use cases that poll remote servers).

We could add one or more new methods in `SftpTransfers` that either return the native JSch `LsEntry` or we could wrapper and modernize those a bit...

**Comment by sleberknight** [comment]
Just realized in that comment that we might not get that information, since it specifically notes "present only if flag SSH_XXX" for each of the items...so maybe not worth it after all...would need to do some investigation and experimentation.

**Reply by terezivy** [reply]
I'd imagine it's useful if those attributes are set more often than not. That said we could probably get those attributes set on the target servers.

**Comment by sleberknight** [comment]
The few services we had that were still using SFTP for anything have now been decommissioned, so this is no longer needed.
